zoloto/marker_type.py

```diff
@@ -1,39 +1,59 @@
 from enum import IntEnum
 
-from cv2 import aruco
+from cv2 import aruco, aruco_Dictionary
 
 
 class MarkerType(IntEnum):
-    ARUCO_4X4_50 = aruco.DICT_4X4_50
-    ARUCO_4X4_100 = aruco.DICT_4X4_100
-    ARUCO_4X4_250 = aruco.DICT_4X4_250
-    ARUCO_4X4_1000 = aruco.DICT_4X4_1000
-    ARUCO_5X5_50 = aruco.DICT_5X5_50
-    ARUCO_5X5_100 = aruco.DICT_5X5_100
-    ARUCO_5X5_250 = aruco.DICT_5X5_250
-    ARUCO_5X5_1000 = aruco.DICT_5X5_1000
-    ARUCO_6X6_50 = aruco.DICT_6X6_50
-    ARUCO_6X6_100 = aruco.DICT_6X6_100
-    ARUCO_6X6_250 = aruco.DICT_6X6_250
-    ARUCO_6X6_1000 = aruco.DICT_6X6_1000
-    ARUCO_7X7_50 = aruco.DICT_7X7_50
-    ARUCO_7X7_100 = aruco.DICT_7X7_100
-    ARUCO_7X7_250 = aruco.DICT_7X7_250
-    ARUCO_7X7_1000 = aruco.DICT_7X7_1000
+    ARUCO_4X4 = aruco.DICT_4X4_1000
+    ARUCO_5X5 = aruco.DICT_5X5_1000
+    ARUCO_6X6 = aruco.DICT_6X6_1000
+    ARUCO_7X7 = aruco.DICT_7X7_1000
     ARUCO_ORIGINAL = aruco.DICT_ARUCO_ORIGINAL
     APRILTAG_16H5 = aruco.DICT_APRILTAG_16H5
     APRILTAG_25H9 = aruco.DICT_APRILTAG_25H9
     APRILTAG_36H10 = aruco.DICT_APRILTAG_36H10
     APRILTAG_36H11 = aruco.DICT_APRILTAG_36H11
 
+    @property
+    def dictionary(self) -> aruco_Dictionary:
+        return aruco.getPredefinedDictionary(self.value)
+
+    @property
+    def marker_count(self) -> int:
+        """
+        The total number of markers available
+        """
+        return len(self.dictionary.bytesList)
+
+    @property
+    def max_id(self) -> int:
+        """
+        The highest id available
+        """
+        return self.marker_count - 1
+
+    @property
+    def min_marker_size(self) -> int:
```

Comment:
Not super happy with the naming here. This is the size just on 1 axis

Reply:
Looks like the aruco paper refers to this as size.

Reply:
I think marker size is the only thing that makes any sense here